jax/_src/numpy/ufuncs.py

```diff
@@ -423,9 +455,44 @@ def expm1(x: ArrayLike, /) -> Array:
   """
   return lax.expm1(*promote_args_inexact('expm1', x))
 
-@implements(np.log1p, module='numpy')
+
 @partial(jit, inline=True)
 def log1p(x: ArrayLike, /) -> Array:
+  """Calculates element-wise logarithm of one plus input, ``log(x+1)``.
+
+  JAX implementation of :obj:`numpy.log1p`.
+
+  Args:
+    x: input array or scalar.
+
+  Returns:
+    An array containing the logarithm of one plus of each element in ``x``,
+    promotes to inexact dtype.
+
+  Note:
+    ``jnp.log1p`` is more accurate than when using the naive computation of
+    ``log(x+1)`` for small values of ``x``.
+
+  See also:
+    - :func:`jax.numpy.expm1`: Calculates :math:`e^x-1` of each element of the
+      input.
+    - :func:`jax.numpy.log2`: Calculates base-2 logarithm of each element of input.
+    - :func:`jax.numpy.log`: Calculates element-wise logarithm of the input.
+
+  Examples:
+    >>> x = jnp.array([2, 5, 9, 4])
+    >>> jnp.allclose(jnp.log1p(x), jnp.log(x+1))
+    Array(True, dtype=bool)
+
+    For values very close to 0, ``jnp.log1p(x)`` is more accurate than
+    ``jnp.log(x+1)``:
+
+    >>> x1 = jnp.array([1e-4, 1e-6, 2e-10])
+    >>> jnp.expm1(jnp.log1p(x1))
+    Array([1.00000005e-04, 9.99999997e-07, 2.00000003e-10], dtype=float32)
+    >>> jnp.expm1(jnp.log(x1+1))
+    Array([1.000166e-04, 9.536743e-07, 0.000000e+00], dtype=float32)
```

Comment:
I'm worried that the detailed numerics here will be system-dependent, and lead to flaky doctest failures. The usual strategy would be to use `set_printoptions` to change the precision, but that would obscure what you're trying to show here, i.e. the detailed precision of each approach. One possible solution would be to use `# doctest: +SKIP` to tell doctest to skip these outputs.

Reply:
Thanks! Done.

Reply:
Wrong line, unfortunately: as written, you will skip the definition of `x1`, and then the doctest will fail with a `NameError` on the next line.

You need to mark the lines with the problematic output with ` # doctest: +SKIP`.

Reply:
Thanks! Understood and modified accordingly.